core/misc/bitset.cpp

```diff
@@ -206,9 +206,7 @@ namespace MR {
     if (have_excess_bits()) {
       if (memcmp (data, that.data, bytes - 1))
         return false;
-      if ((data[bytes - 1] & excess_bit_mask()) != (that.data[bytes - 1] & excess_bit_mask()))
-        return false;
-      return true;
+      return ((data[bytes - 1] & ~excess_bit_mask()) == (that.data[bytes - 1] & ~excess_bit_mask()));
```

Comment:
Can confirm that this issue is specific to the `stats_enhancements` branch; neither `master` not `dev` are affected.